src/transformers/models/superglue/modeling_superglue.py

```diff
@@ -0,0 +1,845 @@
+    ):
+        super().__init__()
+        num_layers = len(channels)
+        layers = []
+        for i in range(1, num_layers):
+            layers.append(nn.Conv1d(channels[i - 1], channels[i], kernel_size=1, bias=True))
+            if i < (num_layers - 1):
+                if do_batch_norm:
+                    layers.append(nn.BatchNorm1d(channels[i]))
+                layers.append(nn.ReLU())
+        self.layers = nn.Sequential(*layers)
+        nn.init.constant_(self.layers[-1].bias, 0.0)
+
+    def forward(
+        self,
+        input: torch.Tensor,
+        output_hidden_states: Optional[bool] = False,
+    ) -> Union[Tuple, torch.Tensor]:
+        all_hidden_states = () if output_hidden_states else None
+        for layer in self.layers:
+            input = layer(input)
+            if output_hidden_states and isinstance(layer, nn.Conv1d):
+                all_hidden_states = all_hidden_states + (input,)
+        output = input
+        return output, all_hidden_states
+
+
+class SuperGlueKeypointEncoder(nn.Module):
+    def __init__(
+        self,
+        config: SuperGlueConfig,
+    ):
+        super().__init__()
+        self.layer_sizes = config.keypoint_encoder_sizes
+        self.feature_dim = config.descriptor_dim
+        self.encoder = SuperGlueMultiLayerPerceptron(config, [3] + self.layer_sizes + [self.feature_dim])
+
+    def forward(
+        self,
+        keypoints: torch.Tensor,
+        scores: torch.Tensor,
+        output_hidden_states: Optional[bool] = False,
+    ) -> torch.Tensor:
+        keypoints = keypoints.transpose(1, 2)
+        scores = scores.unsqueeze(1)
+        inputs = torch.cat([keypoints, scores], dim=1)
+        return self.encoder(inputs, output_hidden_states=output_hidden_states)
+
+
+class SuperGlueMultiHeadAttention(nn.Module):
+    def __init__(self, config: SuperGlueConfig):
+        super().__init__()
+        self.feature_dim = config.descriptor_dim
+        self.num_heads = config.num_heads
+        self.dim = self.feature_dim // self.num_heads
+        self.merge = nn.Conv1d(self.feature_dim, self.feature_dim, kernel_size=1)
+        self.proj = nn.ModuleList([deepcopy(self.merge) for _ in range(3)])
+
+    def forward(
+        self, query: torch.Tensor, key: torch.Tensor, value: torch.Tensor, output_attentions: bool = False
+    ) -> Union[Tuple[torch.Tensor, torch.Tensor], Tuple[torch.Tensor]]:
+        batch_dim = query.size(0)
+        query, key, value = [
+            layer(x).view(batch_dim, self.dim, self.num_heads, -1) for layer, x in zip(self.proj, (query, key, value))
+        ]
+        x, attention_probs = attention(query, key, value)
+        output = self.merge(x.contiguous().view(batch_dim, self.dim * self.num_heads, -1))
+
+        output = (output, attention_probs) if output_attentions else (output,)
+
+        return output
+
+
+class SuperGlueAttentionalPropagation(nn.Module):
+    def __init__(
+        self,
+        config: SuperGlueConfig,
+    ):
```

Comment:
```suggestion
    def __init__(self, config: SuperGlueConfig):
```

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/29886/commits/601da1cf0756cef7339d4dd33faeefbfcd073ac9)